unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -372,23 +382,38 @@ protected void HideAllAgentRenderers()
                                 break;
                             }
                         }
+
+                        //If CapsuleCast check failed, move on to next direction from p (otherwise Check 2's error message could return a false-positive)
+                        if(shouldEnqueue == false) {
+                            continue;
+                        }
+
+                        //Check 2: Is p + d within the bounds of the scene?
                         bool inBounds = agentManager.SceneBounds.Contains(newPosition);
+
                         if (errorMessage == "" && !inBounds) {
                             errorMessage = "In " +
                                 UnityEngine.SceneManagement.SceneManager.GetActiveScene().name +
                                 ", position " + newPosition.ToString() +
                                 " can be reached via capsule cast but is beyond the scene bounds.";
```

Comment:
Why is an error message getting set on this check, but none of the other checks?

Moreover, I'm not convinced that this check is even working properly. If this check is ever true (i.e., `!inBounds`), the `errorMessage` will get set, and based on this line in the `GetReachablePositions` action, no positions will actually ever be returned!
https://github.com/allenai/ai2thor/blob/grid_size_debugging/unity/Assets/Scripts/BaseFPSAgentController.cs#L465

My question is then, are you sure `inBounds` is not always set to true, which might be the case if `agentManager.SceneBounds` contains an infinite space of values.

Reply:
I think that failed return was the point when Luca added this error-check to the logic: to raise the issue.

As for `inBounds` state, that is definitely working as intended. `SceneBounds` has a helper function committed to checking that it isn't infinite, and encompasses the scene so that when p + d reaches outside of it, it's immediately disqualified from being reachable. That being said, this check should hypothetically never need to be used as long as all of our rooms are watertight with collision, which they are now.